### PR TITLE
[2.6] Add CSI 2.4.1 

### PR DESCRIPTION
**Pull request by doflamingo721**
Adds:
CSI v2.4.1
Porting `permit-os` from https://github.com/rancher/charts/pull/1755

Linked issue:
https://github.com/rancher/rancher/issues/33848
rancher/rke2#2619

### CODE REVIEW

File: release.yaml [hunk 1]
```diff
@@ -42,6 +42,7 @@ rancher-pushprox:
 rancher-vsphere-csi:
 - 2.3.1
 - 100.1.1+up2.3.0-rancher1
```
Comment (PennyScissors) [inline release.yaml:44]: Please remove the 100.1.1+up2.3.0-rancher1 entry and its assets, 100.2.0+up2.4.1-rancher1 replaces it.

**Review by pennyscissors — CHANGES_REQUESTED** [review]

File: release.yaml (hunk 1 above)
Comment (doflamingo721) [reply release.yaml]: I made the changes can you please review it again?

**Review by pennyscissors — APPROVED** [review]
LGTM, but adding @rosskirkpat to review this because this PR contains the changes he made in PR https://github.com/rancher/charts/pull/1755 which is not merged yet.

**Comment by PennyScissors** [comment]
Also, make sure you update the new charts issue here: https://github.com/rancher/rancher/issues/36185

**Comment by PennyScissors** [comment]
@rosskirkpat removing you because the changes you made were merged already, so no need to review this anymore unless you want to.

@doflamingo721 could you please rebase this PR and resolve the conflicts? If the conflicts are between the patch in rancher/charts and the windows annotation stuff you ported to rancher/vsphere-charts, the preference is to remove it from the patch in rancher/charts in favor of bringing it from rancher/vsphere-charts.

**Comment by aiyengar2** [comment]
Closing in favor of https://github.com/rancher/charts/pull/1787

**Comment by samkulkarni20** [comment]
@sirredbeard Arvind closed this PR in favor of a [similar change](https://github.com/rancher/charts/pull/1787), but with an additional commit. Hence reopening this PR won't be a good idea. If you need any specific changes from us, pls let us know & we'll raise a new PR for that.
CC: @doflamingo721 @aiyengar2

**Comment by PennyScissors** [comment]
Closing this to avoid confusion since @aiyengar2 already took care of this in PR https://github.com/rancher/charts/pull/1787. Please open a new PR if further changes are needed.

**Pull request closed without merging**